src/hvf/src/lib.rs

```diff
@@ -206,11 +209,27 @@ pub fn vcpu_set_vtimer_mask(vcpuid: u64, masked: bool) -> Result<(), Error> {
     }
 }
 
-pub struct HvfNestedBindings {
-    hv_vm_config_get_el2_supported:
-        libloading::Symbol<'static, unsafe extern "C" fn(*mut bool) -> hv_return_t>,
-    hv_vm_config_set_el2_enabled:
-        libloading::Symbol<'static, unsafe extern "C" fn(hv_vm_config_t, bool) -> hv_return_t>,
+/// Checks if Nested Virtualization is supported on the current system. Only
+/// M3 or newer chips on macOS 15+ will satisfy the requirements.
+pub fn check_nested_virt() -> Result<bool, Error> {
+    type GetEL2Supported =
+        libloading::Symbol<'static, unsafe extern "C" fn(*mut bool) -> hv_return_t>;
+
+    let get_el2_supported: Result<GetEL2Supported, libloading::Error> =
+        unsafe { HVF.get(b"hv_vm_config_get_el2_supported") };
+    if get_el2_supported.is_err() {
+        info!("cannot find hv_vm_config_get_el2_supported symbol");
+        return Ok(false);
+    }
+
+    let mut el2_supported: bool = false;
+    let ret = unsafe { (get_el2_supported.unwrap())(&mut el2_supported) };
+    if ret != HV_SUCCESS {
+        error!("processor does not support the nested virtualization functionality");
```

Comment:
If `hv_vm_config_get_el2_supported` we don't really know if it's supported or not, just that the function call failed. I think we should change this message to reflect that.

Reply:
Should be fixed now.